### PR TITLE
dpres plevel

### PR DESCRIPTION
**Pull request by abb58**
Infrastructure for dpres_plevel

### CODE REVIEW

File: src/geocat/comp/__init__.py [hunk 1]
```diff
@@ -26,6 +26,15 @@ class CoordinateError(Error):
     an argument without a required coordinate array being passed separately."""
     pass
 
+class DimensionError(Error):
+     """Exception raised when the arguments of GeoCAT-comp functions argument
+     has a mismatch of the necessary dimensionality."""
+     pass
+
+class AttrsError(Error):
```
Comment (khallock) [inline src/geocat/comp/__init__.py]: Can we call this `AttributeError` just to be as verbose as the other `Error` classes?

File: src/geocat/comp/_ncomp.pyx [hunk 2]
```diff
@@ -723,3 +723,92 @@ def _moc_globe_atl(np.ndarray lat_aux_grid_np, np.ndarray a_wvel_np, np.ndarray
...
+        then the layer thickness is modifed accordingly. If `psfc` is set to a missing value, all
+        layer thicknesses are set to the appropriate missing value.
+
+        The primary purpose of this function is to return layer thicknesses to be used to
+        weight observations for integrations.
+
+    """
+    plev = Array.from_np(plev_np)
+    psfc = Array.from_np(psfc_np)
+    ptop = Array.from_np(np.ndarray([1], buffer=ptop_scalar, dtype=type(ptop_scalar)))
```
Comment (khallock) [inline src/geocat/comp/_ncomp.pyx]: Is `ptop_scalar` intended to be an optional argument here? As written it's possible for `ptop_scalar=None`, which would result in an invalid type ID (dtype `object`) for an `ncomp_array`.

File: src/geocat/comp/__init__.py (hunk 1 above)
Comment (abb58) [reply src/geocat/comp/__init__.py]: https://github.com/NCAR/geocat-comp/pull/52/commits/53714903ccd172ae5a4f59af5edd97f5413b64ac

File: src/geocat/comp/_ncomp.pyx (hunk 2 above)
Comment (abb58) [reply src/geocat/comp/_ncomp.pyx]: https://github.com/NCAR/geocat-comp/pull/52/commits/5669fe52d8ca7b212090409000f93955819154b8

**Comment by abb58** [comment]
@clyne. Since @khallock have approved now, this PR can safely be merged. @maboualidev and @erogluorhan have already approved this PR before resolving the merge conflicts.

**Comment by khallock** [comment]
@abb58 @clyne I'm merging this since everyone had approved it prior to the new commits. I've also temporarily disabled the "Dismiss stale pull request approvals when new commits are pushed" option to prevent this from happening.

**Pull request merged**